### PR TITLE
Update sendlocation documentation and add a corresponding initializer

### LINKED ISSUE #1170: Update `SendLocation` request RPC documentation
Update the `SendLocation` request RPC documentation to reflect that longitude and latitude are not required (it currently says that they are required). While all parameters are optional, either a longitude/latitude **OR** an address must be provided, otherwise Core will reject the request as invalid data (i.e. either `longitudeDegrees` and `latitudeDegrees` must be set **OR** `address` must be set)

##### OS & Version Information
* iOS Version: n/a
* SDL iOS Version: 6.1
* Testing Against: n/a

### PR DESCRIPTION
**Pull request by joeljfischer**
Fixes #1170 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
Unit tests are added for all initializers

### Summary
Added an address initializer, altered documentation

##### Enhancements
* Added an initializer to `SendLocation` for just using the address and no lat / lon

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: SmartDeviceLink/SDLSendLocation.h [hunk 1]
```diff
@@ -15,21 +15,23 @@ NS_ASSUME_NONNULL_BEGIN
...
+
 - (instancetype)initWithLongitude:(double)longitude latitude:(double)latitude locationName:(nullable NSString *)locationName locationDescription:(nullable NSString *)locationDescription address:(nullable NSArray<NSString *> *)address phoneNumber:(nullable NSString *)phoneNumber image:(nullable SDLImage *)image;
 
 - (instancetype)initWithLongitude:(double)longitude latitude:(double)latitude locationName:(nullable NSString *)locationName locationDescription:(nullable NSString *)locationDescription displayAddressLines:(nullable NSArray<NSString *> *)displayAddressLines phoneNumber:(nullable NSString *)phoneNumber image:(nullable SDLImage *)image deliveryMode:(nullable SDLDeliveryMode)deliveryMode timeStamp:(nullable SDLDateTime *)timeStamp address:(nullable SDLOasisAddress *)address;
 
 /**
- * The longitudinal coordinate of the location.
+ * The longitudinal coordinate of the location. Either the latitude / longitude OR the `addressLines` must be provided.
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLSendLocation.h]: * `addressLines` should be `address` here and in other parts of the code

Comment (joeljfischer) [reply SmartDeviceLink/SDLSendLocation.h]: Fixed

File: SmartDeviceLink/SDLSendLocation.h [hunk 2]
```diff
@@ -48,7 +50,7 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nullable, copy, nonatomic) NSString *locationDescription;
 
 /**
- * Location address for display purposes only
+ * Location address for display purposes only. Either the latitude / longitude OR the `address` must be provided.
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLSendLocation.h]: This needs to be moved to the `address` param documentation.

File: SmartDeviceLink/SDLSendLocation.h [hunk 3]
```diff
@@ -15,21 +15,23 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface SDLSendLocation : SDLRPCRequest
 
+- (instancetype)initWithAddress:(SDLOasisAddress *)address addressLines:(nullable NSArray<NSString *> *)addressLines locationName:(nullable NSString *)locationName locationDescription:(nullable NSString *)locationDescription phoneNumber:(nullable NSString *)phoneNumber image:(nullable SDLImage *)image deliveryMode:(nullable SDLDeliveryMode)deliveryMode timeStamp:(nullable SDLDateTime *)timeStamp;
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLSendLocation.h]: Add documentation to the inits.

**Pull request merged**